src/main/java/org/folio/event/handler/ItemCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,121 @@
+package org.folio.event.handler;
+
+import static org.folio.event.util.KafkaEventUtil.extractTenantFromHeaders;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import java.util.List;
+import java.util.Objects;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.event.DomainEventPayloadType;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.rest.persist.DBClient;
+import org.folio.services.piece.PieceService;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class ItemCreateAsyncRecordHandler extends BaseAsyncRecordHandler<String, String> {
+
+  private static final Logger log = LogManager.getLogger();
+
+  @Autowired
+  private PieceService pieceService;
+
+  public ItemCreateAsyncRecordHandler(Context context, Vertx vertx) {
+    super(vertx, context);
+    SpringContextUtil.autowireDependencies(this, context);
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> kafkaConsumerRecord) {
+    if (log.isDebugEnabled()) {
+      log.debug("handle:: Trying to process kafkaRecord={}", kafkaConsumerRecord.value());
+    }
+
+    var payload = new JsonObject(kafkaConsumerRecord.value());
+
+    String eventType = payload.getString("type");
+    if (!Objects.equals(eventType, DomainEventPayloadType.CREATE.name())) {
+      log.info("handle:: unsupported event type: {}", eventType);
+      return Future.succeededFuture();
+    }
+
+    if (!validatePayload(payload)) {
+      log.warn("handle:: payload validation failed. payload={}", payload);
+      return Future.succeededFuture();
+    }
+
+    try {
+      var tenantId = extractTenantFromHeaders(kafkaConsumerRecord.headers());
+      var dbClient = new DBClient(getVertx(), tenantId);
+      return processItemCreationEvent(payload, dbClient)
+        .onSuccess(v -> log.info("handle:: item '{}' event processed successfully", eventType))
+        .onFailure(t -> log.error("Failed to process event: {}", kafkaConsumerRecord.value(), t))
+        .map(kafkaConsumerRecord.key());
+    } catch (Exception e) {
+      log.error("Failed to process item event kafka record, kafkaRecord={}", kafkaConsumerRecord, e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  private Future<Void> processItemCreationEvent(JsonObject payload, DBClient dbClient) {
+    var tenantId = payload.getString("tenant");
+    var itemObject = payload.getJsonObject("new");
+    var itemId = itemObject.getString("id");
```

Comment:
Can be helpful to move the parameter values to static variables or even better option in an enum. Will be good for reusability

Reply:
Agreee, created enum and dto for these